### PR TITLE
expanded on No Router Instance

### PR DESCRIPTION
**Pull request by therealbigzeez**
Added more detailed explanation of why there is no router instance, and other ways to fix the error

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 9.7s | 9.5s | -221ms |
| nodeModulesSize | 77.6 MB | 77.6 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.908 | 1.918 | ⚠️ +0.01 |
| / avg req/sec | 1310.36 | 1303.56 | ⚠️ -6.8 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.166 | 1.182 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 2143.47 | 2115.14 | ⚠️ -28.33 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| 677f882d2ed8..43e3.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-712e20f..aa05.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-af..9390.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| index.html gzip | 615 B | 615 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 11.7s | 11.6s | -87ms |
| nodeModulesSize | 77.6 MB | 77.6 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| 677f882d2ed8..43e3.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-712e20f..aa05.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-af..9390.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.18 MB | 5.18 MB | ✓ |

</details>

</details>
Commit: 1aa32d49620b0e24e0949a4b73fe5809af6a3373

File: errors/no-router-instance.md [hunk 1]
```diff
@@ -2,8 +2,13 @@
 
 #### Why This Error Occurred
 
+Next.js is universal, which means it executes code first server-side with NodeJS, then on the client-side, `window` object is not defined in NodeJS, so some methods are not supported or availale at build time.
```
Comment (timneutkens) [inline errors/no-router-instance.md:5]:
```suggestion
Next.js is universal, which means it executes code first server-side with NodeJS, then on the client-side, `window` object is not defined in NodeJS, so some methods are not supported or available at build time.
```

The error document is unrelated to `window` not being available 🤔

Comment (Robogeek95) [reply errors/no-router-instance.md:5]: Thanks

Comment (Robogeek95) [reply errors/no-router-instance.md]: I have fixed the typo

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 10s | 10.1s | ⚠️ +43ms |
| nodeModulesSize | 73.3 MB | 73.3 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.987 | 2.024 | ⚠️ +0.04 |
| / avg req/sec | 1257.93 | 1234.99 | ⚠️ -22.94 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.294 | 1.271 | -0.02 |
| /error-in-render avg req/sec | 1931.61 | 1966.77 | +35.16 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| 677f882d2ed8..38ed.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-9ad7be4..1cfb.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-af..9390.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.3s | ⚠️ +70ms |
| nodeModulesSize | 73.3 MB | 73.3 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| 677f882d2ed8..38ed.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-9ad7be4..1cfb.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-af..9390.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.19 MB | 5.19 MB | ✓ |

</details>

</details>
Commit: c1c5a50ab4292b1b4856d0d2ca72cf63989d4de6

**Review by timneutkens — DISMISSED** [review]
tweaked the wording and removed the mention of `window` which does not relate to why the router does not work server-side.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 8.8s | 8.6s | -245ms |
| nodeModulesSize | 73.3 MB | 73.3 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.713 | 1.762 | ⚠️ +0.05 |
| / avg req/sec | 1459.57 | 1419.17 | ⚠️ -40.4 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.095 | 1.107 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 2282.67 | 2257.7 | ⚠️ -24.97 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| 677f882d2ed8..38ed.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-9ad7be4..1cfb.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-af..9390.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 10.4s | 10.2s | -133ms |
| nodeModulesSize | 73.3 MB | 73.3 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| 677f882d2ed8..38ed.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-9ad7be4..1cfb.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-af..9390.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Robogeek95/next.js patch-2 | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.19 MB | 5.19 MB | ✓ |

</details>

</details>
Commit: fe739b1b959a67e48510e2075bca6c24265723b1

**Comment by ijjk** [comment]
## Failing test suites 
Commit: fe739b1b959a67e48510e2075bca6c24265723b1

**test/integration/cli/test/index.test.js** 
- CLI Usage > build > should warn when unknown argument provided
- CLI Usage > start > experimental react version
- CLI Usage > start > experimental react-dom version

<details>
<summary>Expand output</summary>

  ● CLI Usage › build › should warn when unknown argument provided

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 2

    + Browserslist: caniuse-lite is outdated. Please run:
    + npx browserslist@latest --update-db
      Unknown or unexpected option: --random
      ↵

      78 |         stderr: true,
      79 |       })
    > 80 |       expect(stderr).toEqual('Unknown or unexpected option: --random\n')
         |                      ^
      81 |     })
      82 |     test('should not throw UnhandledPromiseRejectionWarning', async () => {
      83 |       const { stderr } = await runNextCommand(['build', '--random'], {

      at Object.<anonymous> (integration/cli/test/index.test.js:80:22)

  ● CLI Usage › start › experimental react version

    expect(received).not.toMatch(expected)

    Expected substring: not "outdated"
    Received string:        "Browserslist: caniuse-lite is outdated. Please run:
    npx browserslist@latest --update-db
    "

      274 | 
      275 |       expect(stderr).not.toMatch('disabled')
    > 276 |       expect(stderr).not.toMatch('outdated')
          |                          ^
      277 |       expect(stderr).not.toMatch(`react-dom`)
      278 | 
      279 |       await killApp(instance)

      at Object.<anonymous> (integration/cli/test/index.test.js:276:26)

  ● CLI Usage › start › experimental react-dom version

    expect(received).not.toMatch(expected)

    Expected substring: not "outdated"
    Received string:        "Browserslist: caniuse-lite is outdated. Please run:
    npx browserslist@latest --update-db
    "

      292 | 
      293 |       expect(stderr).not.toMatch('disabled')
    > 294 |       expect(stderr).not.toMatch('outdated')
          |                          ^
      295 |       expect(stderr).not.toMatch('`react`')
      296 | 
      297 |       await killApp(instance)

      at Object.<anonymous> (integration/cli/test/index.test.js:294:26)

</details>

**Review by timneutkens — APPROVED** [review]
Thanks, I've made some edits and updated the PR.

**Pull request merged**